zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -783,6 +785,130 @@ benchmarks! {
             Ok(())
         })?;
     }: { Pallet::<T>::start_subsidy(&market_clone.unwrap(), market_id)? }
+
+    market_status_manager {
+        let b in 1..31;
+        let f in 1..31;
+
+        // ensure markets exist
+        let start_block: T::BlockNumber = 100_000u64.saturated_into();
+        let end_block: T::BlockNumber = 1_000_000u64.saturated_into();
+        for _ in 0..31 {
+            create_market_common::<T>(
+                MarketCreation::Permissionless,
+                MarketType::Categorical(T::MaxCategories::get()),
+                ScoringRule::CPMM,
+                Some(MarketPeriod::Block(start_block..end_block)),
+            ).unwrap();
+        }
+
+        let range_start: MomentOf<T> = 100_000u64.saturated_into();
+        let range_end: MomentOf<T> = 1_000_000u64.saturated_into();
+        for _ in 31..64 {
+            create_market_common::<T>(
+                MarketCreation::Permissionless,
+                MarketType::Categorical(T::MaxCategories::get()),
+                ScoringRule::CPMM,
+                Some(MarketPeriod::Timestamp(range_start..range_end)),
+            ).unwrap();
+        }
+
+        let block_number: T::BlockNumber = Zero::zero();
+        let last_time_frame: TimeFrame = Zero::zero();
+        for i in 1..=b {
+            <MarketIdsPerOpenBlock<T>>::try_mutate(block_number, |ids| {
+                ids.try_push(i.into())
+            }).unwrap();
+        }
+
+        let last_offset: TimeFrame = last_time_frame + 1.saturated_into::<u64>();
+        //* quadratic complexity should not be allowed in substrate blockchains
+        //* assume at first that the last time frame is one block before the current time frame
+        let t = 0;
```

Comment:
`t` is the number of additional frames?

Reply:
`t` is the time frame span between last time frame and current time frame.